lib/Runtime/Language/Arguments.h

```diff
@@ -4,17 +4,58 @@
 //-------------------------------------------------------------------------------------------------------
 #pragma once
 
+// To extract variadic args array after known args list:
+//      argx, callInfo, ...
+// NOTE: The last known arg name is hard-coded to "callInfo".
 #ifdef _WIN32
-#define VA_LIST_TO_VARARRAY(vl, va, callInfo) Js::Var* va = (Js::Var*) vl;
+#define DECLARE_ARGS_VARARRAY(va, ...)                              \
+    va_list _vl;                                                    \
+    va_start(_vl, callInfo);                                        \
```

Comment:
We better call `va_end` once we are done with it. Previously, I had cleaned up the va leaks. See #1507